proto/accesscontrol/genesis.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+package seiprotocol.seichain.accesscontrol;
+
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/sei-protocol/sei-chain/x/accesscontrol/types";
+
+message GenesisState {
+    Params params = 1 [(gogoproto.nullable) = false];
+}
+  
+enum MessageType {
+    MessageTypeAny = 0;
+}
+
+message MessageDependencyMapping {
+    MessageType message_type = 1;
+    repeated string resource_dependencies = 2;
+}
+
+message Params {
+    option (gogoproto.goproto_stringer) = false;
+    
+    // mapping between every message type and its predetermined resource read/write sequence
+    repeated MessageDependencyMapping message_dependency_mapping = 1 [
```

Comment:
instead of having this be part of params, I think we should have it be its own item managed by accesscontrolkeeper. This way, we can have better key prefixing for each message type so we can get it individually instead of searching for it in a list.

Reply:
Gotcha, thanks for bringing this up.

So I kept it in the params (as we get the governance proposal changes part for free) but then in the keeper added the ability for us to store it based on the key (ModuleName-MessageType), this way the the other modules can still just use the keeper to get the resource dependency individually 

We'll also need to update the gov proposal handler to update each index here